pkg/wal/wal.go

```diff
@@ -43,18 +118,364 @@ type WAL interface {
 	// Write a logging entity.
 	// It will return immediately when the data is written in the buffer,
 	// The returned function will be called when the entity is flushed on the persistent storage.
-	Write(seriesID common.SeriesID, timestamp time.Time, data []byte) (func(), error)
+	Write(seriesID common.SeriesID, timestamp time.Time, data []byte) error
```

Comment:
I found a gap in implementing the Buffer. The seriesID's type should be `[]byte` instead of `common.SeriesID`. 

The reason is that the key in underlying storage comprises the seriesID and tagFamily as https://github.com/apache/skywalking-banyandb/blob/main/banyand/tsdb/series_write.go#L194 indicates.

Reply:
I solve this problem, please review again.